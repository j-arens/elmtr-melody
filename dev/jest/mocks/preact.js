module.exports = {
    unmountComponent: jest.fn(),
};
