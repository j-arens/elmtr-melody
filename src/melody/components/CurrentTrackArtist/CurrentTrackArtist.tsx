import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';

export interface StateProps {
    currentTrackArtist: string;
}

type Props = StateProps & WithOptionalClassName;

export default ({ currentTrackArtist, className = '' }: Props) => (
    <p class={className}>
        {currentTrackArtist}
    </p>
);
