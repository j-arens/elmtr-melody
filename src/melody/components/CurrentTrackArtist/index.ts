import { WithOptionalClassName } from '@components/type';
import { State } from '@redux/type';
import { connect } from 'preact-redux';
import CurrentTrackArtist, { StateProps } from './CurrentTrackArtist';

const mapState = (state: State) => {
    const { audio: { currentTrack, tracks } } = state;
    const activeTrack = tracks[currentTrack];
    return {
        currentTrackArtist: activeTrack.media_details.artist,
    };
};

export default connect<
    StateProps,
    {},
    WithOptionalClassName
>(mapState)(CurrentTrackArtist);
