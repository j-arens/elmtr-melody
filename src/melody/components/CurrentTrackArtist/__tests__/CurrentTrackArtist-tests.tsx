import { h } from 'preact';
import { render } from 'preact-render-to-string';
import CurrentTrackArtist from '../CurrentTrackArtist';

describe('CurrentTrackArtist', () => {
    it('renders', () => {
        const component = render(
            <CurrentTrackArtist
                currentTrackArtist="lol"
                className="signal-green"
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
