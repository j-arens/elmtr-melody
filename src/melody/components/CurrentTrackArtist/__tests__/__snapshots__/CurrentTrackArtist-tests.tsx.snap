// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTrackArtist renders 1`] = `<p class="signal-green">lol</p>`;
